src/component/cyfs-base/src/codec/raw/raw_types.rs

```diff
@@ -757,7 +757,12 @@ impl<T: RawEncode> RawEncode for HashSet<T> {
     ) -> BuckyResult<&'a mut [u8]> {
         let ulen = USize(self.len());
         let mut buf = ulen.raw_encode(buf, purpose)?;
-        for e in self {
+
+        // stable sort
+        let mut values: Vec<&T> = self.iter().collect();
```

Comment:
This should be the change mentioned by https://github.com/buckyos/CYFS/issues/256, it should not be in this pull request

Reply:
Sort the `values` in ascending order, and encode the elements into the buffer. for Issue #256

Reply:
I mean it should be included in a new request for issue #256, and in the entire pull request, it seems, this is the only substantive change?